src/applications/disability-benefits/526EZ/helpers.jsx

```diff
@@ -118,19 +118,21 @@ export function transform(formConfig, form) {
 
   const additionalDocuments = aggregate(disabilities, 'additionalDocuments');
   const privateRecords = aggregate(disabilities, 'privateRecords');
+  const treatments = aggregate(disabilities, 'treatments');
 
   const transformedData = {
     disabilities: disabilities
       .filter(disability => (disability['view:selected'] === true))
       .map(filtered => pick(filtered, disabilityProperties)),
     // Pull phone & email out of phoneEmailCard and into veteran property
     veteran: setPhoneEmailPaths(veteran),
-    // Extract treatments into one top-level array
-    treatments: aggregate(disabilities, 'treatments'),
     attachments: additionalDocuments.concat(privateRecords),
     privacyAgreementAccepted,
     serviceInformation,
     standardClaim,
+    // treatments has a minItems: 1 requirement so only include the property
+    // if there is at least one treatment to send
+    ...treatments.length && { treatments }
```

Comment:
:thinking: How does this spread operator work with `.length`?

Reply:
The whole condition gets evaluated and returns `treatments`, which then gets spread. It's kinda tricky, but I preferred this to a whole bunch of lines that conditionally set up different object shapes based on the `treatments` property.

BTW, `treatments` should always be an array in this case, because the `aggregate` function will return an empty array if it doesn't find any matching properties.

Reply:
So this is more like a `...(treatments.length && { treatments })`? What happens if `!treatments.length`?

Reply:
evaluates to false, which doesn't spread anything. Which is convenient.

Reply:
Cool deal. :+1: